liquibase/liquibase
# Make lock granted time be consistent with date executed time.  Modify listLocks message. LB-2126

## Description
Changes all databases to use the database-based "now" SQL function rather than the client machine's now java code.

This makes it so that the times are always based on the database time and timezone vs. the more variable timezone and times of all the liquibase client machinese.

It also adds a note to the list-locks command to let users know how the date is computed.

- - -
## Dev Handoff Notes (Internal Use)
#### Links
* Fixed Issue: [https://github.com/liquibase/liquibase/issues/2029](https://github.com/liquibase/liquibase/issues/2029) ([https://datical.atlassian.net/browse/LB-2190)](https://datical.atlassian.net/browse/LB-2190))
* Test Results: [https://github.com/liquibase/liquibase/pull/2217/checks](https://github.com/liquibase/liquibase/pull/2217/checks)

#### Testing
* Steps to Reproduce: [https://github.com/liquibase/liquibase/issues/2029](https://github.com/liquibase/liquibase/issues/2029)
* Guidance:
  * Impact: Changes all databases to use the database-based "now" SQL function. The code for what that "now" function is existed before and has been previously tested.

## Test Requirements (Internal Liquibase QA)
Test using Postgres and MySQL.

**Manual Tests**

* Verify update-sql uses the `now()` function to set the DATABASECHANGELOGLOCK.LOCKGRANTED value.
* Verify list-locks shows the lock granted time in the database’s timezone (not client timezone). 

**Automated Tests**

* None required for this fix.



┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-2190) by [Unito](https://www.unito.io)
